src/cli/search.rs

```diff
@@ -93,24 +94,167 @@ pub async fn execute(args: Args) -> miette::Result<()> {
+            .await?;
+    }
+    // If package name filter doesn't contain * (wildcard), it will search and display specific package info (if any package is found)
+    else {
+        let package_name = PackageName::try_from(package_name_filter).into_diagnostic()?;
+        let _ = search_exact_package(package_name, repo_data).await?;
+    }
+
+    Ok(())
+}
+
+async fn search_exact_package(
+    package_name: PackageName,
+    repo_data: Vec<SparseRepoData>,
+) -> miette::Result<()> {
+    let package_name_search = package_name.clone();
+    let packages = await_in_progress(
+        "searching packages",
+        spawn_blocking(move || {
+            search_package_by_filter(&package_name_search, &repo_data, |pn, n| {
+                pn == n.as_normalized()
+            })
+        }),
+    )
+    .await
+    .into_diagnostic()??;
+
+    if packages.is_empty() {
+        let normalized_package_name = package_name.as_normalized();
+        return Err(miette::miette!("Could not find {normalized_package_name}"));
+    }
+
+    let package = packages.last().clone();
+    if let Some(package) = package {
+        print_package_info(package);
+    }
+
+    Ok(())
+}
+
+fn print_package_info(package: &RepoDataRecord) {
+    println!();
+
+    let package = package.clone();
+    let package_name = package.package_record.name.as_source();
+    let build = &package.package_record.build;
+    let package_info = format!("{} {}", console::style(package_name), console::style(build));
+    println!("{}", package_info);
+    println!("{}\n", "-".repeat(package_info.chars().count()));
+
+    println!(
+        "{:19} {:19}",
+        console::style("Name"),
+        console::style(package_name)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("Version"),
+        console::style(package.package_record.version)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("Build"),
+        console::style(build)
+    );
+
+    let size = match package.package_record.size {
+        Some(size) => size.to_string(),
+        None => String::from("Not found."),
+    };
+    println!("{:19} {:19}", console::style("Size"), console::style(size));
+
+    let license = match package.package_record.license {
+        Some(license) => license,
+        None => String::from("Not found."),
+    };
+    println!(
+        "{:19} {:19}",
+        console::style("License"),
+        console::style(license)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("Subdir"),
+        console::style(package.package_record.subdir)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("File Name"),
+        console::style(package.file_name)
+    );
+
+    println!(
+        "{:19} {:19}",
+        console::style("URL"),
+        console::style(package.url)
+    );
+
+    let md5 = match package.package_record.md5 {
+        Some(md5) => hex::encode(md5),
```

Comment:
Why do you need to encode it?

Reply:
I think it is in `bytes`, and encoding turns it into a string. We could also use `format("{:x}", md5)` here, though.

Reply:
Ahh yes indeed, didnt know about this formatting, thanks!